WordPress/Classes/ViewRelated/Post/AztecAttachmentViewController.swift

```diff
@@ -0,0 +1,195 @@
+                alignment = attachment.alignment
+                size = attachment.size
+            }
+        }
+    }
+
+    var alignment = TextAttachment.Alignment.none
+    var size = TextAttachment.Size.full
+
+    fileprivate var handler: ImmuTableViewHandler!
+
+    weak var delegate: AztecAttachmentViewControllerDelegate?
+
+    // MARK: - Initialization
+
+    override init(style: UITableViewStyle) {
+        super.init(style: style)
+        navigationItem.title = NSLocalizedString("Media Settings", comment: "Media Settings Title")
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    required convenience init() {
+        self.init(style: .grouped)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        ImmuTable.registerRows([
+            EditableTextRow.self,
+            ], tableView: self.tableView)
+
+        handler = ImmuTableViewHandler(takeOver: self)
+        handler.viewModel = tableViewModel()
+
+        WPStyleGuide.configureColors(for: view, andTableView: tableView)
+
+        let cancelButton = UIBarButtonItem(barButtonSystemItem: .cancel, target:self, action: #selector(AztecAttachmentViewController.handleCancelButtonTapped))
+        navigationItem.leftBarButtonItem = cancelButton
+
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target:self, action: #selector(AztecAttachmentViewController.handleDoneButtonTapped))
+        navigationItem.rightBarButtonItem = doneButton
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.handler.viewModel = self.tableViewModel()
+    }
+
+    // MARK: - Model mapping
+
+    func tableViewModel() -> ImmuTable {
+        let displaySettingsHeader = NSLocalizedString("Web Display Settings", comment: "The title of the option group for editing an image's size, alignment, etc. on the image details screen.")
+
+        let alignmentRow = EditableTextRow(
+            title: NSLocalizedString("Alignment", comment: "Image alignment option title."),
+            value: alignTitles[alignment]!,
+            action: displayAlignmentSelector)
+
+        let sizeRow = EditableTextRow(
+            title: NSLocalizedString("Size", comment: "Image size option title."),
+            value: sizeTitles[size]!,
+            action: displaySizeSelector)
+
+        return ImmuTable(sections: [
+            ImmuTableSection(
+                headerText: displaySettingsHeader,
+                rows: [
+                    alignmentRow,
+                    sizeRow,
+                ],
+                footerText: nil)
+            ])
+    }
+
+
+    // MARK: - Actions
+
+    func displayAlignmentSelector(row: ImmuTableRow) {
+
+        let values = alignTitles.map { (key: TextAttachment.Alignment, value: String) -> TextAttachment.Alignment in
+            return key
+        }
```

Comment:
Note here! (No need to change anything). This snippet can be condensed into something like this:

```
        let values = alignTitles.map { (key, value) in
            return key
        }
```

- No need to specify the **alignTitles** `key/value` types, since those are known
- The return type can be also inferred by the compiler